tests/e2e/upgrade_test.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright © 2022 - 2025 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_test
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	"github.com/rancher-sandbox/ele-testhelpers/rancher"
+	"github.com/rancher-sandbox/ele-testhelpers/tools"
+)
+
+var _ = Describe("E2E - Upgrading Rancher Manager", Label("upgrade-rancher-manager"), func() {
+	// Create kubectl context
+	// Default timeout is too small, so New() cannot be used
+	k := &kubectl.Kubectl{
+		Namespace:    "",
+		PollTimeout:  tools.SetTimeout(300 * time.Second),
+		PollInterval: 500 * time.Millisecond,
+	}
+
+	It("Upgrade Rancher Manager", func() {
+
+		// Get before-upgrade Rancher Manager version
+		getImageVersion := []string{
+			"get", "pod",
+			"--namespace", "cattle-system",
+			"-l", "app=rancher",
+			"-o", "jsonpath={.items[*].status.containerStatuses[*].image}",
+		}
+		versionBeforeUpgrade, err := kubectl.RunWithoutErr(getImageVersion...)
+		Expect(err).To(Not(HaveOccurred()))
+
+		// Get Fleet Version before-upgrade
+		getFleetImageVersion := []string{"get", "pod",
+		"--namespace", "cattle-fleet-system",
+		"-l", "app=fleet-controller",
+		"-o", "jsonpath={.items[*].status.containerStatuses[*].image}",
+		}
+
+		// Execute the shell command to get version before upgrade
+		fleetVersionBeforeUpgrade, err := kubectl.RunWithoutErr(getFleetImageVersion...)
+
+		// Upgrade Rancher Manager
+		// NOTE: Don't check the status, we can have false-positive here...
+		//       Better to check the rollout after the upgrade, it will fail if the upgrade failed
+		_ = rancher.DeployRancherManager(
+			rancherHostname,
+			rancherUpgradeChannel,
+			rancherUpgradeVersion,
+			rancherUpgradeHeadVersion,
+			"None",
+			"None",
+		)
+
+		// Wait for Rancher Manager to be restarted
+		// NOTE: 1st or 2nd rollout command can sporadically fail, so better to use Eventually here
+		Eventually(func() string {
+			status, _ := kubectl.RunWithoutErr(
+				"rollout",
+				"--namespace", "cattle-system",
+				"status", "deployment/rancher",
+			)
+			return status
+		}, tools.SetTimeout(4*time.Minute), 30*time.Second).Should(ContainSubstring("successfully rolled out"))
```

Comment:
nit: Isn't 30s a long polling interval? Setting it to 10s, for instance, as done in checks below, could shorten some test runs.

Reply:
yeah, it could be real quicker and will not wait 20s more. I will update it.